docs/external-disk.md

```diff
@@ -19,45 +19,108 @@ The `-disk` specification may be repeated for multiple disks, although a limited
+The final (optional) argument specifies the device name
 
-1. format container
-    * The format container needs to have bind mounts for `/dev`
-    * The format container needs `CAP_SYS_ADMIN` and `CAP_MKNOD` capabilities
-    * The format container only needs to run **once**, not matter how many external disks or partitions are provided. It finds all block devices under `/dev` and processes them.
-    * The default container config should be sufficient
-2. mount container
-    * The mount container `command` is `mount.sh` followed by the desired mount point. Remember that nearly everything in a linuxkit image is read-only except under `/var`, so mount it there.
-    * The mount container needs to have bind mounts for `/dev` and `/var`
-    * The mount container needs `CAP_SYS_ADMIN` capabilities
-    * The mount container needs `rootfsPropagation: shared`
-    * The default container config should be sufficient, though the `mount.sh` command needs to be specified
+## Mount the disk
 
-With the above in place, if run with the current disk options, the image will make the external disk available as `/dev/vda1` and mount it at `/var/external`.
+Once a disk has been prepared it will need to be mounted using `pkg/mount`
+
+### Usage
+
+**NOTE: Block devices may only be mounted in `/var` unless you have explicitly added an additional bind mount**
+
+In this mode of operation, the first unmounted linux partition on the any block device is mounted to the mountpoint provided.
+
+```
+onboot:
+  name: mount
+  image: linuxkit/format:${hash}
+  command: ["/usr/bin/mountie", "/var/lib/docker"]
+```
+
+### Options
+
+You can provide either a partition label, device name or disk UUID to specify which disk should be used.
+For example:
+
+```
+onboot:
+  name: format
+  image: linuxkit/format:${hash}
+  command: ["/usr/bin/mountie", "-label", "DATA", "/var/lib/docker" ]
+```
+
+```
+onboot:
+  name: format
+  image: linuxkit/format:${hash}
+  command: ["/usr/bin/mountie", "-uuid", "a-proper-uuid", "/var/lib/docker" ]
+```
+
+```
+onboot:
+  name: format
+  image: linuxkit/format:${hash}
+  command: ["/usr/bin/mountie", "-device", "/dev/sda1", "/var/lib/docker" ]
+```
+
+For compatibility with the standard `mount` command we also support providing the device name as a positional argument.
+E.g
+
+```
+onboot:
+  name: format
+  image: linuxkit/format:${hash}
+  command: ["/usr/bin/mountie", "/dev/sda1", "/var/lib/docker" ]
+```
+
+## Extending Partitions
+
+`pkg/extend` can extends a single partition to fill the entire disk
+
+### Usage
+
+In the default mode of operation, any disks that are found and have a single partition and free space will have that partition extended.
+
+```
+onboot:
+  name: extend
+  image: linuxkit/extend:${hash}
+```
+
+If you know the name of the disk that you wish to extend you may supply this as an argument
+
+```
+onboot:
+  name: extend
+  image: linuxkit/extend:${hash}
+  command: ["/usr/bin/extend", "/dev/vda"]
```

Comment:
It kind of seems more natural to specify `/dev/vda1` to me as that is what is extended though its a bit weird either way...